bl-kernel/helpers/sanitize.class.php

```diff
@@ -30,10 +30,14 @@ public static function htmlDecode($text)
 		return htmlspecialchars_decode($text, $flags);
 	}
 
-	// Remove javacript from links
+	// Remove javascript from links
 	public static function noJSLink($text)
 	{
-		return preg_replace("/javascript\s*:\s*/", "", $text);
+		$text = trim($text);
+		while(strpos($text, 'javascript')===0){
```

Comment:
Checking for `javascript` only could lead to `invalid replacement` and subsequent `dos` issue:

![Screenshot from 2020-08-31 11-48-59](https://user-images.githubusercontent.com/33063403/91707537-3e319700-eb80-11ea-9a7a-ad40f152a7e6.png)

PoC used:

```php
<?php

function noJSLink($text) {
 $text = trim($text); 
 while(strpos($text, 'javascript')===0){
    $text = preg_replace("/javascript\s*:\s*/", "", $text); 
 } 
return $text; 
}

echo noJSLink('javascript:javascript');

?>
```

I would check for `javascript:` in the `while` :smile:

Reply:
Done, thanks